docs/i18n.md

```diff
@@ -27,47 +28,122 @@ NODE_APP_INSTANCE=[MY_APP] NODE_PATH='./:./src' bin/create-locales
+Start the process by creating a git branch and extracting the locales for a
+given app. This uses `amo` as an example app but you would need to repeat the
+process in a new branch for `disco` and any other activate application
+(example: `NODE_APP_INSTANCE=disco ...`).
+
+```
+git checkout master
+git pull
+git checkout -b amo-locales
+NODE_APP_INSTANCE=amo bin/extract-locales
+```
+
+This extracts all strings wrapped with `i18n.gettext()` or any other
+function supported by [Jed][Jed] (the library we use in JavaScript to carry out
 replacements for the string keys in the current locale).
 
 The strings are extracted using a babel plugin via webpack. Extracted strings
 are added to a pot template file. This file is used to seed the po for each
 locale with the strings needing translating when merging locales.
 
-To extract + merge new strings for an app in the addons-frontend project
-you should run the following commands:
+Run `git diff` to see what the extraction did.
+**If no strings were updated then you do not have to continue creating the pull
+request. You can revert the changes made to the `pot` timestamp.**
+Here is an example of a diff where no strings were changed. It just shows
+a single change to the timestamp:
+
+```diff
+diff --git a/locale/templates/LC_MESSAGES/amo.pot b/locale/templates/LC_MESSAGES/amo.pot
+index 31e113f2..c7da4e34 100644
+--- a/locale/templates/LC_MESSAGES/amo.pot
++++ b/locale/templates/LC_MESSAGES/amo.pot
+@@ -2,7 +2,7 @@ msgid ""
+ msgstr ""
+ "Project-Id-Version: amo\n"
+ "Report-Msgid-Bugs-To: EMAIL@ADDRESS\n"
+-"POT-Creation-Date: 2017-06-08 14:01+0000\n"
++"POT-Creation-Date: 2017-06-08 14:43+0000\n"
+ "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+ "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+ "Language-Team: LANGUAGE <LL@li.org>\n"
+```
+
+When the application is under active development it's more likely that you will
+see a diff containing new strings or at least strings that have shifted to
+different line numbers in the source. If so, commit your change and continue
+to the next step:
+
+```
+git commit -a "Extract AMO locales"
+```
+
+### Merging locale files
+
+After extracting new strings, you have to merge them into the existing locale
+files. Do this in your branch and commit:
 
 ```
-# First extract the new strings.
-NODE_APP_INSTANCE=[MY_APP] bin/extract-locales
-# Merge the new strings into the locale files.
-NODE_APP_INSTANCE=[MY_APP] bin/merge-locales
+NODE_APP_INSTANCE=amo bin/merge-locales
+```
+
+At this point, you may want to check for
+[fuzzy strings](https://www.gnu.org/software/gettext/manual/html_node/Fuzzy-Entries.html)
+by running `git diff` and searching for a comment that looks like
+`# fuzzy`. This comment means the localization may not exactly match the source text;
+a localizer needs to review it.
+For applications like `disco` (The Discovery Pane) you may not want
+to merge a pull request until the fuzzy strings are resolved.
```

Comment:
You really have to merge them or localizers won't see them / or fix them.

I think the main points are about checking that the fuzzy markers are expected. E.g. you'd expect to see them when you extract a new string that's a partial match to a key that already exists.

Developers should look out for where code changes cause translations to be marked fuzzy e.g. by changing the copy or changes to the extraction code. In exceptional cases you may prefer to remove fuzzy markers manually to avoid falling back to an english string even if the resulting translation isn't exact any more. 

Removing the fuzzy marker manually will make it less obvious to translators that the translation needs updating so use that option with care and or contact the localizers mailing list to let them know.

An important point is that a fuzzy marked translations will *not* be used in the application by default. This is controlled by config. Allowing fuzzy translations would almost always result in bad translations being used in the application so turning that on should be done with caution.